internal/infrastructure/kubernetes/deployment.go

```diff
@@ -0,0 +1,232 @@
+	if err != nil {
+		if kerrors.IsNotFound(err) {
+			deploy, err := i.createDeployment(ctx, infra)
+			if err != nil {
+				return err
+			}
+			if err := i.addResource(deploy); err != nil {
+				return err
+			}
+			return nil
+		}
+		return err
+	}
+
+	if err := i.addResource(current); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// getDeployment gets the Deployment for the provided infra from the kube api.
+func (i *Infra) getDeployment(ctx context.Context, infra *ir.Infra) (*appsv1.Deployment, error) {
+	ns := i.Namespace
+	name := infra.Proxy.Name
+	key := types.NamespacedName{
+		Namespace: ns,
+		Name:      infra.GetProxyInfra().ObjectName(),
+	}
+	deploy := new(appsv1.Deployment)
+	if err := i.Client.Get(ctx, key, deploy); err != nil {
+		return nil, fmt.Errorf("failed to get deployment %s/%s: %w", ns, name, err)
+	}
+
+	return deploy, nil
+}
+
+// expectedDeployment returns the expected Deployment based on the provided infra.
+func (i *Infra) expectedDeployment(infra *ir.Infra) *appsv1.Deployment {
+	containers := expectedContainers(infra)
+
+	deployment := &appsv1.Deployment{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "Deployment",
+			APIVersion: "apps/v1",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: i.Namespace,
+			Name:      infra.GetProxyInfra().ObjectName(),
+			Labels:    envoyLabels(),
+		},
+		Spec: appsv1.DeploymentSpec{
+			Replicas: pointer.Int32(1),
+			Selector: EnvoyPodSelector(),
+			Template: corev1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: EnvoyPodSelector().MatchLabels,
+				},
+				Spec: corev1.PodSpec{
+					Containers: containers,
+					Volumes: []corev1.Volume{
+						{
+							Name: envoyCfgVolName,
+							VolumeSource: corev1.VolumeSource{
+								EmptyDir: &corev1.EmptyDirVolumeSource{},
+							},
+						},
+					},
+					ServiceAccountName:            infra.Proxy.ObjectName(),
+					AutomountServiceAccountToken:  pointer.BoolPtr(false),
+					TerminationGracePeriodSeconds: pointer.Int64Ptr(int64(300)),
+					SecurityContext:               newUnprivilegedPodSecurity(),
+					DNSPolicy:                     corev1.DNSClusterFirst,
+					RestartPolicy:                 corev1.RestartPolicyAlways,
+					SchedulerName:                 "default-scheduler",
+				},
+			},
+		},
+	}
+
+	return deployment
+}
+
+func expectedContainers(infra *ir.Infra) []corev1.Container {
+	ports := []corev1.ContainerPort{
+		{
+			Name:          "http",
+			ContainerPort: envoyHTTPPort,
+			Protocol:      corev1.ProtocolTCP,
+		},
+		{
+			Name:          "https",
+			ContainerPort: envoyHTTPSPort,
+			Protocol:      corev1.ProtocolTCP,
+		},
+	}
+
+	containers := []corev1.Container{
+		{
+			Name:            envoyContainerName,
+			Image:           infra.Proxy.Image,
+			ImagePullPolicy: corev1.PullIfNotPresent,
+			Command: []string{
+				"envoy",
+			},
+			Args: []string{
+				"-c",
+				filepath.Join("/", envoyCfgVolMntDir, envoyCfgFileName),
```

Comment:
nit: IMO in 2022 short flags are only valuable as shorthand for interactively typing commands in a terminal.  When writing code, use the long form (`--config-path`) for readability, same as we use more-than-one-character variable names.